Model/Cron.php

```diff
@@ -355,7 +355,27 @@ public function execute()
         $notifications->addFieldToFilter('created_at', $dateRange);
         $notifications->addFieldToFilter('error_count', ['lt' => Notification::MAX_ERROR_COUNT]);
 
+        // Process the notifications in ascending order by creation date and event_code
+        $notifications->getSelect()->order('created_at ASC')->order('event_code ASC');
```

Comment:
If the OFFER_CLOSED and AUTHORISATION success = true notification arrives in the same second and the cron job is not running for more than 10 minutes and it is restarted manually then it will makes sure that the authorisation is processed before the offer_closed. So you can see it is a really really edge case but can happen.

Reply:
I see now, thanks